service/history/historyEngine.go

```diff
@@ -748,3 +749,40 @@ func (e *historyEngineImpl) GetReplicationStatus(
 ) (_ *historyservice.ShardReplicationStatus, retError error) {
 	return replicationapi.GetStatus(ctx, request, e.shard, e.replicationAckMgr)
 }
+
+func (e *historyEngineImpl) ForceDeleteWorkflowExecution(
```

Comment:
I think you don't need a new API for this. The replication task executor can call DeleteWorkflowExecution API. It doesn't matter is the deletion is async or not.